File: 2015/rust/01/part_one.rs
/*
 * Advent of Code 2015 day 01 part one
 */

use std::fs::File;
use std::io::BufReader;

// String slice constant
const INPUT_FILE: &str = "../inputs/01-input.txt";

fn main() {
    let mut file = BufReader::new(
        File::open(INPUT_FILE).expect(&format!("Failed to open file '{}'", INPUT_FILE)),
    );
}

File: 2015/C/01/part_one.c
/*
 * Advent of Code 2015 day 01 part one
 */

#include <stdio.h> // fopen, fclose, fgetc, fprintf, printf
#include <stdlib.h> // exit

#define INPUT_FILE "../inputs/01-input.txt"

int main()
{
    int floor = 0;
    unsigned int index = 0;
    char ch = '\0';
    FILE *file = fopen(INPUT_FILE, "r");

    if (file == NULL)
    {
        fprintf(stderr, "Failed to open file '%s'\n", INPUT_FILE);
        exit(EXIT_FAILURE);
    }

    while ((ch = fgetc(file)) != EOF)
    {
        switch (ch)
        {
            case '(':
                floor++;
                break;

            case ')':
                floor--;
                break;

            default:
                fprintf(stderr, "Invalid character found at position '%u'\n", index + 1);
                fclose(file);
                exit(EXIT_FAILURE);
        }
        index++;
    }

    printf("The right floor is '%d'\n", floor);

    fclose(file);
    exit(EXIT_SUCCESS);
}
